docs/developer/extending/webhooks/synchronous-events/stored-payment-method.mdx

```diff
@@ -186,3 +186,240 @@ The response in this case should have the following structure:
+
+The webhook payload includes details about the [customer](api-reference/payments/objects/payment-method-initialize-tokenization-session.mdx#code-style-fontweight-normal-paymentmethodinitializetokenizationsessionbuserbcodeuser--), [channel](api-reference/payments/objects/payment-method-initialize-tokenization-session.mdx#code-style-fontweight-normal-paymentmethodinitializetokenizationsessionbchannelbcodechannel--), and [data](api-reference/payments/objects/payment-method-initialize-tokenization-session.mdx#code-style-fontweight-normal-paymentmethodinitializetokenizationsessionbdatabcodejson-). The [data](api-reference/payments/objects/payment-method-initialize-tokenization-session.mdx#code-style-fontweight-normal-paymentmethodinitializetokenizationsessionbdatabcodejson-) field contains the JSON data provided as input in the [paymentMethodInitializeTokenization](api-reference/payments/mutations/payment-method-initialize-tokenization.mdx) mutation.
+
+The response must conform to the expected format and will be returned within the [paymentMethodInitializeTokenization](api-reference/payments/mutations/payment-method-initialize-tokenization.mdx) mutation.
+
+### Request
+
+Saleor will send a [PAYMENT_METHOD_INITIALIZE_TOKENIZATION_SESSION](api-reference/webhooks/enums/webhook-event-type-sync-enum.mdx#code-style-fontweight-normal-webhookeventtypesyncenumbpayment_method_initialize_tokenization_sessionbcode) webhook using the `PaymentMethodInitializeTokenizationSession` subscription type or with a predefined payload in case of a missing subscription query.
+
+For detailed information on building webhook subscription queries, please refer to [this documentation](../../../extending/webhooks/subscription-webhook-payloads).
+
+The following example demonstrates a sample webhook subscription defined to handle the tokenization of a payment method:
+
+```graphql
+subscription {
+  event {
+    ... on PaymentMethodInitializeTokenizationSession {
+      paymentFlowToSupport
+      user {
+        id
+      }
+      channel {
+        id
+      }
+      data
+    }
+  }
+}
+```
+
+The next example illustrates the predefined payload that will be used if a subscription query is not provided:
+
+```json
+{
+  "user_id": "VXNlcjoyOA==",
+  "channel_slug": "main",
+  "data": { "data": "from-input" },
+  "payment_flow_to_support": "interactive"
+}
+```
+
+### Response
+
+- A response indicating successful tokenization should include the `result`, the `id` of the stored payment method, and optionally `data` to be passed to the storefront:
+
+```json
+{
+  "result": "SUCCESSFULLY_TOKENIZED",
+  "id": "payment-method-id",
+  "data": { "data": "additional-data" }
+}
+```
+
```

Comment:
Shouldn't we have 4th result here when the tokenization is pending? E.g. Adyen can return success or pending response when tokenizing credit card

Reply:
Added